src/hooks/useTasks.js

```diff
@@ -39,36 +39,42 @@ function useOpenTasks(tasks, now, courtSettings) {
           currentRoundPhases[i].phase === DisputesTypes.Phase.AppealRuling ||
```

Comment:
So these changes looks good, but the `useMemo()` may look simpler / more clear rewritten as:

```js
useMemo(() => {
  if (!tasks) {
    return []
  }

  const incompleteTasks = tasks.filter((_, i) => currentRoundPhases[i].phase !== DisputesTypes.Phase.Ended)
  const openTasks = []
  for (let i = 0; i < incompleteTasks.length; i++) {
    const currentPhase = currentRoundPhases[i].phase

    if (
      currentPhase !== DisputesTypes.Phase.AppealRuling &&
      currentPhase !== DisputesTypes.Phase.ConfirmAppeal
    ) {
      for (let j = 0; j < tasks[i].jurors.length; j++) {
        if (
          isVotingTaskOpen(
            tasks[i].jurors[j],
            currentPhase
          )
        ) {
          openTasks.push({
            number: tasks[i].number,
            state: tasks[i].state,
            createdAt: parseInt(tasks[i].createdAt, 10) * 1000,
            juror: tasks[i].jurors[j].juror.id,
            disputeId: tasks[i].dispute.id,
            commitment: tasks[i].jurors[j].commitment,
            outcome: tasks[i].jurors[j].outcome,
            phase: getTaskName(currentRoundPhases[i].phase),
            phaseType: currentRoundPhases[i].phase,
            dueDate: currentRoundPhases[i].nextTransition,
            open: true,
          })
        }
      }
    } else if (isAppealTaskOpen(tasks[i], currentPhase)) {
      // We are in appeal or confirm and only need to generate a single task
      // (rather than one per juror) if the task is still open
      openTasks.push({
        number: tasks[i].number,
        state: tasks[i].state,
        createdAt: parseInt(tasks[i].createdAt, 10) * 1000,
        juror: 'Anyone',
        disputeId: tasks[i].dispute.id,
        phase: getTaskName(currentRoundPhases[i].phase),
        dueDate: currentRoundPhases[i].nextTransition,
        phaseType: currentRoundPhases[i].phase,
        open: true,
      })
    }
  }

  return openTasks
}
```

Notice how the indenting has been significantly reduced, and we could "lift" the nested ifs out.

(And to double check, on the `currentPhase !== DisputesTypes.Phase.Ended`; this _is_ something we can just filter out at the beginning? If a task's phase is ended, it should never be considered, right?)